### PR TITLE
ARROW-2227: [Python] Fix off-by-one error in chunked binary conversions

### PR DESCRIPTION
**Pull request by wesm**
We were already testing the chunked behavior but we did not exercise the off-by-one error edge case

### CODE REVIEW

File: cpp/src/arrow/python/numpy_to_arrow.cc [hunk 1]
```diff
@@ -60,7 +60,7 @@ namespace py {
 
 using internal::NumPyTypeSize;
 
-constexpr int64_t kBinaryMemoryLimit = std::numeric_limits<int32_t>::max();
+constexpr int64_t kBinaryMemoryLimit = std::numeric_limits<int32_t>::max() - 1;
```
Comment (pitrou) [inline cpp/src/arrow/python/numpy_to_arrow.cc]: Apparently this is already available as `BinaryBuilder::kMaximumCapacity` (though for some reason it's protected).

Comment (wesm) [reply cpp/src/arrow/python/numpy_to_arrow.cc]: I can move this constexpr to builder.h and use that consistently

Comment (wesm) [reply cpp/src/arrow/python/numpy_to_arrow.cc]: done

**Comment by wesm** [comment]
Appveyor build: https://ci.appveyor.com/project/wesm/arrow/build/1.0.1770. Merging

**Pull request closed without merging**